homeassistant/helpers/integration_platform.py

```diff
@@ -12,23 +13,35 @@
 from homeassistant.setup import ATTR_COMPONENT
 
 _LOGGER = logging.getLogger(__name__)
+DATA_INTEGRATION_PLATFORMS = "integration_platforms"
 
 
-@bind_hass
-async def async_process_integration_platforms(
-    hass: HomeAssistant,
-    platform_name: str,
-    # Any = platform.
-    process_platform: Callable[[HomeAssistant, str, Any], Awaitable[None]],
+@dataclass(frozen=True)
+class IntegrationPlatform:
+    """An integration platform."""
+
+    platform_name: str
+    process_platform: Callable[[HomeAssistant, str, Any], Awaitable[None]]
+    processed_platforms: set[str]
+
+
+async def async_process_integration_platform(
+    hass: HomeAssistant, component_name: str
 ) -> None:
-    """Process a specific platform for all current and future loaded integrations."""
+    """Process component being loaded or on demand."""
+    if "." in component_name:
+        return
 
-    async def _process(component_name: str) -> None:
-        """Process component being loaded."""
-        if "." in component_name:
-            return
+    integration = await async_get_integration(hass, component_name)
 
-        integration = await async_get_integration(hass, component_name)
+    integration_platforms: list[IntegrationPlatform] = hass.data[
+        DATA_INTEGRATION_PLATFORMS
+    ]
+    for integration_platform in integration_platforms:
+        platform_name = integration_platform.platform_name
+        if component_name in integration_platform.processed_platforms:
+            continue
+        integration_platform.processed_platforms.add(component_name)
```

Comment:
I don't think we know yet here that this component has a platform of the kind that the `integration_platform` item represents.

Reply:
I was thinking we shouldn't try to process it again later if the import fails which is why I always add it to the `processed_platforms` `set()` before trying anything (and explicitly testing for not calling it twice)

I'm not 100% sure thats what you mean though.

Reply:
I could rename `processed_platforms` to `seen_platforms` instead?

Reply:
007010c13c6236a75c2ce12e9e92cb481a5c29d8

Reply:
Ok, sounds good.